### PR TITLE
Tic 0.47.0

### PR DESCRIPTION
**Pull request by matheuslessarodrigues**
Beside things described in issue299 ([link](https://github.com/nesbox/TIC-80/issues/299)), I've moved the `memset(buffer, 0, TIC_CODE_SIZE);` to inside the `if(contents)`.

Also, in order to implement the "-watch" option, I decided to do it as one single command "-code-watch".

So in order to invoke TIC injecting external code, one would still use
```
tic -code code.lua
```
And if you also want to watch for file modifications on window focus (sdl event)
```
tic -code-watch code.lua
```
I found this to be simpler as I didn't have to change this bit of code in `console.c`
```c
if (argc == 2) cmdLoadCart(console, argv[1]);
else if (argc == 3)
{
	cmdInjectCode(console, argv[1], argv[2]);
	cmdInjectSprites(console, argv[1], argv[2]);
	cmdInjectMap(console, argv[1], argv[2]);
}
else if (argc > 3)
{
	cmdLoadCart(console, argv[1]);

	for (s32 i = 2; i < argc; i += 2)
	{
		cmdInjectCode(console, argv[i], argv[i + 1]);
		cmdInjectSprites(console, argv[i], argv[i + 1]);
		cmdInjectMap(console, argv[i], argv[i + 1]);
	}
}
```
I belive that once we need to code more complex command line options, we'd be better with some command line arg parsing lib.

Is it cool? :)

### CODE REVIEW

File: src/console.c [hunk 1]
```diff
@@ -2317,7 +2317,8 @@ static void tryReloadCode(Console* console, char* codeBuffer)
 
 static void cmdInjectCode(Console* console, const char* param, const char* name)
 {
-	if(strcmp(param, "-code") == 0)
+	bool watch = strcmp(param, "-code-watch") == 0;
+	if(strcmp(param, "-code") == 0 || watch)
```
Comment (incinirate) [inline src/console.c]: Micro-optimization: format as `if(watch || strcmp(param, "-code") == 0)` to prevent doing 2 `strcmp`s when only 1 is needed.

Comment (matheuslessarodrigues) [reply src/console.c]: Fair enough!

**Comment by nesbox** [comment]
Pls don't forget to update the Wiki https://github.com/nesbox/TIC-80/wiki#command-line-arguments
Thanks

**Comment by nesbox** [comment]
@matheuslessarodrigues 
found a bug here #346
pls fix

**Comment by matheuslessarodrigues** [comment]
noted. should fix latter today!

**Pull request merged**